File: docs/examples/go/secure_message.go
package main

import (
	"encoding/base64"
	"fmt"
	"github.com/cossacklabs/themis/gothemis/keys"
	"github.com/cossacklabs/themis/gothemis/message"
	"os"
)

func main() {
	if 5 != len(os.Args) {
		fmt.Printf("usage: %s <command> <private key> <peer public key> <message>\n", os.Args[0])
		return
	}
	decodedKey, err := base64.StdEncoding.DecodeString(os.Args[2])
	if nil != err {
		fmt.Println("error decoding private key")
		return
	}
	pr := keys.PrivateKey{Value: decodedKey}
	decodedKey, err = base64.StdEncoding.DecodeString(os.Args[3])
	if nil != err {
		fmt.Println("error decoding private key")
		return
	}
	pu := keys.PublicKey{Value: decodedKey}
	sm := message.New(&pr, &pu)
	if "enc" == os.Args[1] {
		wrapped, err := sm.Wrap([]byte(os.Args[4]))
		if nil != err {
			fmt.Println("error encrypting message")
			return
		}
		fmt.Println(base64.StdEncoding.EncodeToString(wrapped))
	} else if "dec" == os.Args[1] {
		decodedMessage, err := base64.StdEncoding.DecodeString(os.Args[4])
		if nil != err {
			fmt.Println("error decoding message")
			return
		}
		unwrapped, err := sm.Unwrap(decodedMessage)
		if nil != err {
			fmt.Println("error decrypting message")
			return
		}
		fmt.Println(string(unwrapped[:]))

	}
}

File: docs/examples/ts/secure_message.ts
import { SecureMessage } from 'jsthemis';

if (process.argv.length == 6) {
    const command = process.argv[2];
    const private_key = process.argv[3];
    const peer_public_key = process.argv[4];

    const secure_message = new SecureMessage(
        Buffer.from(private_key, "base64"),
        Buffer.from(peer_public_key, "base64"));

    if (command == "enc") {
        const encrypted_message = secure_message.encrypt(Buffer.from(process.argv[5]));
        console.log(Buffer.from(encrypted_message).toString("base64"));
    } else if (command == "dec") {
        const decrypted_message = secure_message.decrypt(Buffer.from(process.argv[5], "base64"));
        console.log(Buffer.from(decrypted_message).toString("utf8"));
    } else {
        console.log("usage node secure_message.js <enc/dec> <private key> <peer public key> <message>");
    }
} else {
    console.log("usage node secure_message.js <enc/dec> <private key> <peer public key> <message>");
}